### PR TITLE
adding support for PyPortal and a simpletest example.

### PR DESCRIPTION
**Pull request by FoamyGuy**
Someone from discord was trying to use the pybadger library with a pyportal recently and it didn't work because it wasn't meant to be supported by this library. 

I was able to add support for PyPortal as well as created and tested a modified simpletest example that works on the PyPortal with touchscreen to advance between examples instead of physical buttons.

The PyPortal has no physical buttons like the rest of the devices this library supports.

### CODE REVIEW

**Comment by fede2cr** [comment]
Tested the branch on my pyportal with circuitpython 5.4 beta, and it works excellent.
I even tried the dimmer from the simple example and it works fine.

Thanks you so much for you help with this.

File: adafruit_pybadger/pyportal.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    @property
+    def _unsupported(self):
+        """This feature is not supported on PyPortal."""
+        raise NotImplementedError("This feature is not supported on PyPortal.")
+
+    # The following is a list of the features available in other PyBadger modules but
+    # not available for PyPortal. If called while using a PyPortal, they will result in the
+    # NotImplementedError raised in the property above.
+    play_file = _unsupported
```
Comment (jerryneedell) [inline adafruit_pybadger/pyportal.py]: Why is play_file unsupported?

File: adafruit_pybadger/pyportal.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    light = _unsupported
```
Comment (jerryneedell) [inline adafruit_pybadger/pyportal.py]: They PyPortal also has a light sensor. Can't it be supported?

Comment (fede2cr) [reply adafruit_pybadger/pyportal.py]: Weird. The following line does dim down the screen. I've tested it in a pybadge and the behavior seems the same. 

pybadger.auto_dim_display()

Note: The dim code acts weird by itself. If you change the environment of *both* boards, they don't seem to react to it. It looks like it only changes the backlight once, even if it's on the loops. But seems like something for another bug request.

File: adafruit_pybadger/pyportal.py (hunk 1 above)
Comment (FoamyGuy) [reply adafruit_pybadger/pyportal.py]: It wasn't left unsupported for a specific reason. I started the PyPortal class with a copy of the existing CLUE class and it got brought in as unsupported from there. I just didn't have time to work on adding it after the display stuff was working before I went to bed last night. 

I will add some more commits today to add the other functionalities that are possible on this hardware. 

Looking back in the CLUE file now it is setting up an AudioPWMOut object but then setting play_file as unsupported. That seems odd to me at first glance, but maybe there is a reason. I'll poke around a bit today.

File: adafruit_pybadger/pyportal.py (hunk 2 above)
Comment (FoamyGuy) [reply adafruit_pybadger/pyportal.py]: Yes I think we can support the light sensor. I will work on adding that today. 

I think the auto_dim functionality uses the accelerometer in the PyBadge to check for movement and dim the display when there hasn't been any. The PyPortals do not have one so we won't be able to have it work the same way, but perhaps could be adapted to use touch or something else.

**Comment by FoamyGuy** [comment]
`play_file` and `light` were actually mostly working already they just needed to not get set to _unsupported and tweak the pin for the light sensor. 

The latest commits fix those and I also added a new example blinka bmp file that fits the PyPortal screen a bit better.

Thank you jerry and fede2cr for the feedback.

**Comment by FoamyGuy** [comment]
#34 should be taken in favor of this. It contains the PyPortal support and example as well as some refactoring to it that took place while working on #34.

If it's preferred to have them separated let me know and we can re-open this one.

**Pull request closed without merging**